app/src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,148 @@
+package duke;
+
+import java.time.format.DateTimeParseException;
+
+public class Duke implements InputAcceptor {
+    private Ui ui;
+    private Storage storage;
+    private Parser parser;
+    private TaskList tasks;
+
+    /**
+     * Creates a new Duke chatbot.
+     */
+    public Duke(String fileName) {
+        storage = new Storage(fileName);
+        parser = new Parser();
+        tasks = storage.load();
+    }
+
+    public void setUi(Ui ui) {
+        this.ui = ui;
+    }
+
+    /**
+     * Runs the Duke chatbot.
+     */
+    public void run() {
+        ui.respond("Hello! You have " + tasks.size() + " tasks. What can I do for you today?");
+
+        ui.runInputLoop();
+    }
+
+    @Override
+    public void input(String input) {
+        String[] splits = parser.splitOnFirst(input, " ");
+        ui.respond(handle(splits[0], splits[1]));
+    }
+
+    private int checkTask(String id) {
+        try {
+            int task = Integer.parseInt(id);
+            if (task > tasks.size() || task <= 0) {
+                return -1;
+            }
+            return task;
+        } catch (NumberFormatException e) {
+            return -2;
+        }
+    }
+
+    private String handle(String command, String params) {
```

Comment:
Good application of making the happy path prominent. Makes it easy to see what should happen should nothing go wrong. Nice!